docs/reference/contributing/guidelines/workflow.md

```diff
@@ -79,6 +79,43 @@ Each feature has a tech lead. This person is responsible for:
 - Rebasing often to track master development.
 - Reviewing any addition to the feature branch (approval required by the feature tech lead or another assigned person).
 
+##### Pull request types
+
+We consider the following pull request types.
+
+##### Fix
+
+A bug fix should be backward-compatible internal change that fixes incorrect behavior. The fix should provide a description what and how it fixes the issue. The test should be included to catch this issue in the future and confirm the correct behavior.
+
+Release: patch version
+
+##### Refactor
+
+Refactors are intended for minor versions if they are not fixing specific issue as they 
+can introduce new issues.
+
+Release: minor version
+
+##### New target
+
+Adding a new target targets the patch release as it updating targets folder implementation.
+
+Release: patch version
+
+##### Feature
+
+New features targets minor releases. It can be integrated only if the feature supports most of the targets (if it requires new target HAL implementation).
```

Comment:
Echoing @geky comment about what is a major/minor/patch release, if it isn't already documented.